src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/CompleteNextDataSetVersionImportFunctionTests.cs

```diff
@@ -0,0 +1,365 @@
+                .DataSetVersionImports
+                .AsNoTracking()
+                .Single(import => import.DataSetVersionId == nextVersion.Id);
+
+            var result = await CompleteNextDataSetVersionImport(
+                dataSetVersionId: nextVersion.Id,
+                durableTaskClientMock.Object);
+
+            VerifyAllMocks(durableTaskClientMock);
+
+            var responseViewModel = result.AssertOkObjectResult<ProcessDataSetVersionResponseViewModel>();
+
+            // Assert that the pre-existing import entry for the next data set version is re-used.
+            var updatedDataSetVersionImport = publicDataDbContext
+                .DataSetVersionImports
+                .Single(import => import.DataSetVersionId == nextVersion.Id);
+
+            Assert.Equal(originalDataSetVersionImport.Id, updatedDataSetVersionImport.Id);
+
+            // Assert that the InstanceId has been set to a new unique value.
+            Assert.NotEqual(originalDataSetVersionImport.InstanceId, updatedDataSetVersionImport.InstanceId);
+
+            // Assert that the InstanceId returned in the view model is the new one rather than the original.
+            Assert.Equal(updatedDataSetVersionImport.InstanceId, responseViewModel.InstanceId);
+
+            // Assert the processing orchestrator was scheduled with the correct arguments
+            Assert.NotNull(processNextDataSetVersionContext);
+            Assert.NotNull(startOrchestrationOptions);
+            Assert.Equal(new ProcessDataSetVersionContext { DataSetVersionId = nextVersion.Id },
+                processNextDataSetVersionContext);
+            Assert.Equal(
+                new StartOrchestrationOptions { InstanceId = updatedDataSetVersionImport.InstanceId.ToString() },
+                startOrchestrationOptions);
+        }
+
+        [Fact]
+        public async Task DataSetVersionIdIsEmpty_ReturnsValidationProblem()
+        {
+            var result = await CompleteNextDataSetVersionImport(
+                dataSetVersionId: Guid.Empty);
+
+            var validationProblem = result.AssertBadRequestWithValidationProblem();
+
+            validationProblem.AssertHasNotEmptyError(
+                nameof(DataSetVersionProcessRequest.DataSetVersionId).ToLowerFirst());
+        }
+
+        [Fact]
+        public async Task DataSetVersionIsNotFound_ReturnsValidationProblem()
+        {
+            var result = await CompleteNextDataSetVersionImport(
+                dataSetVersionId: Guid.NewGuid());
+
+            var validationProblem = result.AssertBadRequestWithValidationProblem();
+
+            validationProblem.AssertHasError(
+                expectedPath: nameof(DataSetVersionProcessRequest.DataSetVersionId).ToLowerFirst(),
+                expectedCode: ValidationMessages.DataSetVersionNotFound.Code);
+        }
+
+        [Fact]
+        public async Task DataSetVersionNotInMappingStatus_ReturnsValidationProblem()
```

Comment:
could turn this into a `[Theory]` and test every other Status

Reply:
What a fine idea, changed to a theory ta.